### ISSUE TITLE
Assignment operator, `=`, not in a group

### ISSUE THREAD

**Issue by toastal** [nickel-lang/tree-sitter-nickel#32]

**Comment by vkleen** [comment]
Nickel doesn't really have a proper assignment operator. The `=` is treated as a syntactic separator in record literals, let bindings and pattern matches. That being said, is the current tree-sitter structure giving you trouble with formulating any queries in particular? We could certainly imagine tweaking the tree structure to make queries easier to construct.

**Comment by toastal** [comment]
That was my questions is how to properly differentiate `=` since the operated is overloaded with many meanings.

**Comment by yannham** [comment]
Is there any motivation for doing so, such as analysis, highlighting, formatting (as in topiary)? Or is it more like a general question around what would be the best way of organizing the grammar?

**Comment by toastal** [comment]
Yes. I’m using Tree-sitter + Neovim + `conceal` to turn operators into something that is clearer for me to understand at a glance. Overloaded `=` isn’t as clear, so I’d like to swap one of them to a different symbol.

**Comment by toastal** [comment]
I can put together a query that works, but I was kinda curious in general. I’ve looked at some other grammars out there & it’s really not consistent in any I checked whether or not “assignment operators” are labeled separately. But I started with looking at Nickel for my config since Nickel was the language I was most recently working with, so I ended up here.

**Comment by yannham** [comment]
In general, I don't know if there are good resource about the idiomatic way of writing a tree-sitter grammar. For Nickel, what we did was mostly to copy one-for-one the rule from the LALRPOP grammar, which should be not too far from a standard EBNF grammar definition (with the restriction, well, that the grammar is LALR(1), albeit we cheat a bit during tokenization). Then we did a few tweaks to make it work with tree-sitter, and latter on a bit more to make it easy for Topiary to format Nickel code, but I must admit we haven't really thought through the tree-sitter users queries side of the story...
